lib/src/main/java/com/basistheory/android/service/ProxyApi.kt

```diff
@@ -0,0 +1,109 @@
+package com.basistheory.android.service
+
+import com.basistheory.ApiClient
+import com.basistheory.android.model.ElementValueReference
+import com.basistheory.android.util.isPrimitiveType
+import com.basistheory.android.util.toMap
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import java.lang.reflect.Type
+
+interface Proxy {
+
+    suspend fun get(proxyRequest: ProxyRequest, apiKeyOverride: String? = null): Any?
+
+    suspend fun post(proxyRequest: ProxyRequest, apiKeyOverride: String? = null): Any?
+
+    suspend fun put(proxyRequest: ProxyRequest, apiKeyOverride: String? = null): Any?
+
+    suspend fun patch(proxyRequest: ProxyRequest, apiKeyOverride: String? = null): Any?
+
+    suspend fun delete(proxyRequest: ProxyRequest, apiKeyOverride: String? = null): Any?
+}
+
+enum class HttpMethod {
+    GET,
+    POST,
+    PATCH,
+    PUT,
+    DELETE
+}
+
+class ProxyRequest {
+    var path: String? = null
+    var queryParams: Map<String, String>? = emptyMap()
+    var headers: Map<String, String>? = emptyMap()
+    var body: Any? = null
+}
+
+class ProxyApi(val apiClientProvider: (apiKeyOverride: String?) -> ApiClient) : Proxy {
+
+    private val ioDispatcher: CoroutineDispatcher = Dispatchers.IO
+
+    override suspend fun get(proxyRequest: ProxyRequest, apiKeyOverride: String?): Any? =
+        withContext(ioDispatcher) {
+            proxy(HttpMethod.GET.name, proxyRequest, apiKeyOverride)
+        }
+
+    override suspend fun post(proxyRequest: ProxyRequest, apiKeyOverride: String?): Any? =
+        withContext(ioDispatcher) {
+            proxy(HttpMethod.POST.name, proxyRequest, apiKeyOverride)
+        }
+
+    override suspend fun put(proxyRequest: ProxyRequest, apiKeyOverride: String?): Any? =
+        withContext(ioDispatcher) {
+            proxy(HttpMethod.PUT.name, proxyRequest, apiKeyOverride)
+        }
+
+    override suspend fun patch(proxyRequest: ProxyRequest, apiKeyOverride: String?): Any? =
+        withContext(ioDispatcher) {
+            proxy(HttpMethod.PATCH.name, proxyRequest, apiKeyOverride)
+        }
+
+    override suspend fun delete(proxyRequest: ProxyRequest, apiKeyOverride: String?): Any? =
+        withContext(ioDispatcher) {
+            proxy(HttpMethod.DELETE.name, proxyRequest, apiKeyOverride)
+        }
+
+    private fun proxy(method: String, proxyRequest: ProxyRequest, apiKeyOverride: String?): Any? {
+        val apiClient = apiClientProvider(apiKeyOverride)
+
+        val call = apiClient.buildCall(
+            "https://api.basistheory.com/proxy",
```

Comment:
We should pass the apiUrl and dispatcher into the ProxyApi constructor from BasisTheoryElements, so that we use the same configured values here (it is a best practice I read to always allow library clients to pass in their own dispatchers).